tests/unit/services/spotify.spec.js

```diff
@@ -0,0 +1,92 @@
+import * as SpotifyService from '@/services/spotify'
+
+const withAdapter = async (axios, token, fn) => {
+    const request = jest.fn()
+        .mockImplementation(() => Promise.resolve({ data: null }))
+
+    const original = axios.defaults.adapter
+    axios.defaults.adapter = request
+
+    const configWith = options => ({
```

Comment:
Not a huge fan of this - axios has a handful of default request configuration options which made it difficult to do a direct assertion on what the config looked like as a request.

Now that I'm looking at this, we could just record a collection of the config objects, which would make it easier to do something like `expect(config[0].url).toBe('...')` rather than abuse jest mock functions.